extensions/amp-ad-network-triplelift-impl/0.1/test/test-amp-ad-network-triplelift-impl.js

```diff
@@ -0,0 +1,129 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpAdNetworkTripleliftImpl} from '../amp-ad-network-triplelift-impl';
+import {AmpAdUIHandler} from '../../../amp-ad/0.1/amp-ad-ui'; // eslint-disable-line no-unused-vars
+import {
+ AmpAdXOriginIframeHandler,    // eslint-disable-line no-unused-vars
+} from '../../../amp-ad/0.1/amp-ad-xorigin-iframe-handler';
+import {base64UrlDecodeToBytes} from '../../../../src/utils/base64';
+import {utf8Encode} from '../../../../src/utils/bytes';
+import * as sinon from 'sinon';
+import {tripleliftIsA4AEnabled} from '../triplelift-a4a-config';
+import {createElementWithAttributes} from '../../../../src/dom';
+import {createIframePromise} from '../../../../testing/iframe';
+
+describe.only('triplelift-a4a-config', () => {
+  let doc;
+  let win;
+  beforeEach(() => {
+    return createIframePromise().then(f => {
+      doc = f.doc;
+      win = f.win;
+    });
+  })
+  it('should pass a4a config predicate', () => {
+    const element = createElementWithAttributes(doc, 'amp-ad', {
+      src: 'https://ib.3lift.com/ttj?inv_code=ampforadstest_main_feed',
+      'data-use-a4a': 'true'
+    });
+    expect(tripleliftIsA4AEnabled(win, element)).to.be.true;
+  });
+  it('should fail a4a config predicate due to missing use-a4a', () => {
+    const element = createElementWithAttributes(doc, 'amp-ad', {
+      src: 'https://ib.3lift.com/ttj?inv_code=ampforadstest_main_feed',
+    });
+    expect(tripleliftIsA4AEnabled(win, element)).to.be.false;
+  });
+  it('should fail a4a config predicate due to missing src', () => {
+    const element = createElementWithAttributes(doc, 'amp-ad', {
+      'data-use-a4a': 'true'
+    });
+    expect(tripleliftIsA4AEnabled(win, element)).to.be.false;
+  });
+  it('should fail a4a config predicate due to invalid src', () => {
+    const element = createElementWithAttributes(doc, 'amp-ad', {
+      src: 'https://evil.com?hello=world&https://ib.3lift.com',
+      'data-use-a4a': 'true'
+    });
+    expect(tripleliftIsA4AEnabled(win, element)).to.be.false;
+  });
+});
+
+describe('amp-ad-network-triplelift-impl', () => {
+
+  let sandbox;
+  let tripleliftImpl;
+  let tripleliftImplElem;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+    tripleliftImplElem = document.createElement('amp-ad');
+    tripleliftImplElem.setAttribute('type', 'triplelift');
+    tripleliftImplElem.setAttribute('src', 'https://ib.3lift.com/ttj?inv_code=ampforadstest_main_feed');
+    tripleliftImplElem.setAttribute('data-use-a4a','true');
+    sandbox.stub(AmpAdNetworkTripleliftImpl.prototype, 'getSigningServiceNames',
+        () => {
+          return ['cloudflare'];
+        });
+    tripleliftImpl = new AmpAdNetworkTripleliftImpl(tripleliftImplElem);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('#isValidElement', () => {
```

Comment:
added the negative test; but again,, since we’re checking for src attribute and base url in tripleliftIsA4AEnabled is there a reason to check it again here?